FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,47 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 	
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
+		client.get(from: url, completion: { [ weak self ] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case .success((let data, let response)):
+				completion(FeedImageMapper.map(from: data, response))
+			}
+		})
+	}
+	
+	private class FeedImageMapper {
+		static let OK_200: Int = 200
+		
+		static func map(from data: Data, _ response: HTTPURLResponse) -> FeedLoader.Result {
+			if response.statusCode != OK_200 {
+				return .failure(Error.invalidData)
+			} else {
+				let decoder = JSONDecoder()
+				do {
+					let root = try decoder.decode(Root.self, from: data)
+					return .success(root.items.map { $0.feedImage })
+				} catch  {
+					return .failure(Error.invalidData)
+				}
+			}
```

Comment:
You can simplify the implementation of this function by grouping the branches returning `.failure(Error.invalidData)`:

```suggestion
			guard response.statusCode == OK_200, let root = try? JSONDecoder().decode(Root.self, from: data) else {
				return .failure(Error.invalidData)
			}
			
			return .success(root.items.map { $0.feedImage })
```

Reply:
Awesome, thanks, updated.

Reply:
Nice one!